lib/con_cache.ex

```diff
@@ -115,7 +117,18 @@ defmodule ConCache do
   Higher values put less pressure on processing, but item expiry is less precise.
   """
   @spec start_link(options, [name: GenServer.name]) :: Supervisor.on_start
-  def start_link(options \\ [], sup_opts \\ []) do
+  def start_link(options, sup_opts \\ []) do
+    case {Keyword.fetch(options, :ttl), Keyword.fetch(options, :ttl_check)} do
```

Comment:
In general, I have a hard time reading this function. Currently `start_link` is more concerned about validation, while its main task (starting) is pushed into a separate private function, which is sometimes invoked other times isn't. I find it hard to tell forrest from the trees here :-)

I suggest refactoring this into something like:

```elixir
with :ok <- validate_ttl(options[:ttl], options[:ttl_check]), do:
  Supervisor.start_link(
    # ...
  )
```

Where `validate_ttl` would be a multiclause.

Notice that I'm not using `fetch` here. I don't think we need fetch, since `nil` is not a valid value anyway, so using access would improve the readability of `validate_ttl` multiclause, since we don't need to match on `{:ok, _} | :error`.